lib/Test/MockTime/HiRes.pm

```diff
@@ -80,7 +80,7 @@ sub gettimeofday() {
     if (defined $Test::MockTime::fixed) {
         return wantarray ? do {
             my $int_part = int($Test::MockTime::fixed);
-            ($int_part, 1_000_000 * ($Test::MockTime::fixed - $int_part))
+            ($int_part, 1_000_000 * sprintf('%.6f', ($Test::MockTime::fixed - $int_part)))
```

Comment:
Now I have another question.  My REPL for Perl 5.18.2 says that the value of `$result` in the following code is `1000000`.  Is this the intended behavior?

```perl
my $t = 1.9999999999
my $result = 1_000_000 * sprintf("%.6f", $t - int($t))
```